src/wp-includes/media.php

```diff
@@ -1897,6 +1911,15 @@ function wp_filter_content_tags( $content, $context = null ) {
 		_prime_post_caches( $attachment_ids, false, true );
 	}
 
+	/**
+	 * Filters whether the auto sizes enabled for lazy load images.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param bool $is_auto_sizes_enabled Whether the auto sizes enabled for lazy load images. Default true.
+	 */
+	$is_auto_sizes_enabled = apply_filters( 'wp_img_auto_sizes_enabled', true );
```

Comment:
If we add a filter to opt out of this feature ([pending this discussion](https://github.com/WordPress/wordpress-develop/pull/7253#issuecomment-2322007086)), I think a better location would be inside the `wp_img_tag_add_auto_sizes()` function, and have that function return the original `sizes` value if the filter is set to `false`.